File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        l = 0
        for r in range(1,len(nums)):
            if nums[l] ==0:
                nums[l], nums[r] = nums[r],nums[l]
                if nums[l]!=0:
                    l+=1
            else:
                l+=1
        print(nums)

File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int i = 0; // Tracks the position for the next non-zero element
        
        for (int j = 0; j < nums.length; j++) {
            if (nums[j] != 0) {
                // Swap elements at i and j
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                
                // Move the non-zero boundary forward
                i++;
            }
        }
    }
}